empire.go

```diff
@@ -602,8 +642,12 @@ func (opts DeployOpts) Validate(e *Empire) error {
 
 // Deploy deploys an image and streams the output to w.
 func (e *Empire) Deploy(ctx context.Context, opts DeployOpts) (*Release, error) {
+	if err := authorizeApp(ctx, "Deploy", opts.App); err != nil {
```

Comment:
An easier interim solution might be to to allow a nil resource. If there's an explicit deny statement, and if the provided resource is nil, then the action is not allow, regardless of the resource constraint. e.g.:

```go
var policy = acl.Policy{
        acl.Statement{
                Effect: acl.Deny,
                Action: []string{"Deploy"},
                Resource: []string{"app-name"},
        },
}

allowed := policy.Allow(acl.Context{
        Action: "Deploy",
        Resource: nil, // unknown
})
fmt.Println(allowed)
// Output: false
```

I think a more correct, and long term solution, would be to refactor so that authorizations are checked on an internal api (could just be lowercase versions of methods to keep things simple).